Modules/_threadmodule.c

```diff
@@ -71,6 +75,51 @@ get_thread_state_by_cls(PyTypeObject *cls)
     return get_thread_state(module);
 }
 
+// Helper to set the thread name using platform-specific APIs
+static int
+set_native_thread_name(const char *name)
+{
+#ifdef __APPLE__
+    return pthread_setname_np(name);
+#elif defined(__NetBSD__)
+    pthread_t thread = pthread_self();
+    return pthread_setname_np(thread, "%s", (void *)name);
+#elif defined(HAVE_PTHREAD_SETNAME_NP)
+    pthread_t thread = pthread_self();
+    return pthread_setname_np(thread, name);
+#elif defined(HAVE_PTHREAD_SET_NAME_NP)
+    pthread_t thread = pthread_self();
+    pthread_set_name_np(thread, name);
+    return 0; /* pthread_set_name_np() returns void */
+#else
+    return 0;
+#endif
```

Comment:
```suggestion
    int rc;
#ifdef __APPLE__
    rc = pthread_setname_np(name);
#elif defined(__NetBSD__)
    pthread_t thread = pthread_self();
    rc = pthread_setname_np(thread, "%s", (void *)name);
#elif defined(HAVE_PTHREAD_SETNAME_NP)
    pthread_t thread = pthread_self();
    rc = pthread_setname_np(thread, name);
#elif defined(HAVE_PTHREAD_SET_NAME_NP)
    pthread_t thread = pthread_self();
    pthread_set_name_np(thread, name);
    rc = 0; /* pthread_set_name_np() returns void */
#endif
    return rc;
```

I think it's easier if we only have one return statement here but that's up to you.